lib/Doctrine/MongoDB/Aggregation/Expr.php

```diff
@@ -281,17 +281,22 @@ public function cond($if, $then, $else)
     }
 
     /**
-     * Ensures an array or operator expression is converted to an array.
+     * Converts an expression object into an array, recursing into nested items
+     *
+     * For expression objects, it calls getExpression on the expression object.
+     * For arrays, it recursively calls itself for each array item. Other values
+     * are returned directly
      *
      * @param mixed|self $expression
-     * @return mixed
+     * @return string|array
+     * @internal
      */
-    protected function ensureArray($expression)
+    public static function convertExpression($expression)
```

Comment:
Yes, that's the main reason for the refactoring: it's used in a couple of the new stages, so to avoid duplicating the behavior I decided to make it a public method instead. It's marked internal since there's no need to use it in applications.